### ISSUE TITLE
Feature: (sort-classes) split static property/method group into static and static-private

### ISSUE THREAD

**Issue by ArchitektApx** [azat-io/eslint-plugin-perfectionist#67]
### What rule do you want to change?

sort-classes

### Describe the problem

Thank you for fixing #62. 

## tl;dr
Now that private and public methods are properly differentiated I'd suggest splitting the current static methods and static properties into static public and static private. 

## reasoning
One might ask why? Simple: to align the behaviour of the static property and method groups with the other property and method groups as described below.

## details
Current groups behave differently for instance public/private:
- **instance properties**:
  - `private-property` for _/#/private properties
  - `property` for public properties (or all properties if private/static is not used)
- **instance methods**:
  - `private-method` for privates
  - `methods` for publics (or all methods if private/static is not used)
 
versus static public/private:
- **static properties**:
  - `static-property` for both private and public
- **static methods**:
  - `static-method` both private and public

The supposed change would align the behaviour of static and instance groups for methods and properties:
- **static properties**:
  - `static-private-property` for privates
  - `static-property` for publics (or all static properties if static private is not used)
- **static methods**:
  - `static-private-method` for privates
  - `static-method` for publics (or all static methods if static private is not used)

### Code example

this would allow configs like this:

```json
"perfectionist/sort-classes": [
      "error",
      {
        "type": "natural",
        "order": "asc",
        "groups": [
          "static-property",
          "static-private-property",
          "property",
          "private-property",
          "constructor",
          "method",
          "private-method",
          "static-method",
          "static-private-method",
          "other"
        ]
      }
    ],
```

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Added in 37512c9a63299aad25c522a58c785770f11547bc.
Will be released today.
